packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -33,58 +28,134 @@ export interface ColumnHeaderProps<T extends string = string> extends PageCounte
      * Ignored when allowedOrderTypes is defined
      */
     withUndefined?: boolean;
+    allowedOrderTypes?: Array<OrderType>;
+};
+
+export type HasSortColumn<T> =
+    | (ColumnInfo<T> & {options?: undefined})
+    | ({
+          title: string;
+          shortTitle?: string;
+          column: T;
+          options: Array<ColumnInfo<T>>;
+      } & Partial<Record<keyof Omit<ColumnInfo<T>, 'column' | 'title' | 'shortTitle'>, undefined>>);
 
-    loading?: boolean;
+export type ColumnHeaderProps<T extends string = string> = PageCounterProps &
+    HasSortColumn<T> & {
+        className?: string;
+        order?: OrderType;
+        multisortIndex?: number;
+        onSort?: (
+            column: T,
+            nextOrder: OrderType,
+            options: {currentOrder?: OrderType; multisort?: boolean},
+        ) => void;
+
+        loading?: boolean;
+
+        align?: 'center' | 'left' | 'right';
+    };
+
+type NameTitleContent = {
+    nameContent: string;
+    titleContent: string;
+};
+
+function getNameTitle<T extends string>({
+    column,
+    title,
+    shortTitle,
+}: Pick<ColumnInfo<T>, 'column' | 'title' | 'shortTitle'>): NameTitleContent {
+    const t = title ?? _.capitalize(column);
+    const nameContent = !shortTitle ? t : shortTitle;
+    return {nameContent, titleContent: t};
+}
+
+function useColumnInfo<T extends string = string>(
+    props: ColumnHeaderProps<T>,
+): ColumnInfo<T> & NameTitleContent & {subColumn?: NameTitleContent} {
+    if (props.options === undefined) {
+        const {column, allowUnordered, withUndefined, title, shortTitle, allowedOrderTypes} = props;
+        return {
+            column,
+            allowUnordered,
+            withUndefined,
+            allowedOrderTypes,
+            ...getNameTitle({column, title, shortTitle}),
+        };
+    } else {
```

Comment:
"else" is not needed. Above you do "return"

Reply:
Such isolated `if`-s (without `else`) looks like prevention of some unwanted behaviour below. But here `if/else`-expression is spliting of code-execution for two different types of input props of `HasSortColumn` (it is more like typeguard):
```ts
if (props.options === undefined) {
   // handling of type (ColumnInfo<T> & {options?: undefined})
} else {
  // handling of type 
  // ({
  //        title: string;
  //        shortTitle?: string;
  //        column: T;
  //        options: Array<ColumnInfo<T>>;
  //    } & Partial<Record<keyof Omit<ColumnInfo<T>, 'column' | 'title' | 'shortTitle'>, undefined>>)
}
```
The current form works the same as your suggestion but it is more expressive.

Reply:
fixed